### PR TITLE
store, kvencoder: add a options to disable txn local latches in mockTikvStore, which may cause lightning to be oom.

### PR DESCRIPTION
**Pull request by winkyao**
Lightning use mockTikvStore to convert sql to kv pairs, it is newed by NewTestTiKVStore:

```
// NewTestTiKVStore creates a test store with Option
func NewTestTiKVStore(client Client, pdClient pd.Client, clientHijack func(Client) Client, pdClientHijack func(pd.Client) pd.Client, disableTxnLocalLatches bool) (kv.Storage, error) {
	if clientHijack != nil {
		client = clientHijack(client)
	}

	pdCli := pd.Client(&codecPDClient{pdClient})
	if pdClientHijack != nil {
		pdCli = pdClientHijack(pdCli)
	}

	// Make sure the uuid is unique.
	uid := uuid.NewV4().String()
	spkv := NewMockSafePointKV()
	tikvStore, err := newTikvStore(uid, pdCli, spkv, client, false)
	tikvStore.EnableTxnLocalLatches(1024000)

	tikvStore.mock = true
	return tikvStore, errors.Trace(err)
}
```

NewTestTiKVStore will definitely new a 100mb txn local latches, which is useless in lightning, and will lead lightning to oom. So we add an option to disable it in kvencoder.

@holys @huachaohuang PTAL

### CODE REVIEW

**Comment by holys** [comment]
LGTM

**Comment by huachaohuang** [comment]
LGTM

File: store/mockstore/tikv.go [hunk 1]
```diff
@@ -80,6 +81,13 @@ func WithPath(path string) MockTiKVStoreOption {
 	}
 }
 
+// WithTxnLocalLatches enable or disble txnLocalLatches.
+func WithTxnLocalLatches(enable bool) MockTiKVStoreOption {
```
Comment (shenli) [inline store/mockstore/tikv.go]: Why not call it SetTxnLocalLatches?

Comment (winkyao) [reply store/mockstore/tikv.go]: We already names previous function in format `Withxxxx`:

```
// WithHijackClient hijacks KV client's behavior, makes it easy to simulate the network
// problem between TiDB and TiKV.
func WithHijackClient(wrap func(Client) Client) MockTiKVStoreOption {
	return func(c *mockOptions) {
		c.clientHijack = wrap
	}
}

// WithCluster provides the customized cluster.
func WithCluster(cluster *mocktikv.Cluster) MockTiKVStoreOption {
	return func(c *mockOptions) {
		c.cluster = cluster
	}
}

// WithMVCCStore provides the customized mvcc store.
func WithMVCCStore(store mocktikv.MVCCStore) MockTiKVStoreOption {
	return func(c *mockOptions) {
		c.mvccStore = store
	}
}
```

**Pull request merged**